pkg/raft/node_test.go

```diff
@@ -375,7 +375,7 @@ func TestNodeProposeWaitDropped(t *testing.T) {
 		}
 		n.Advance()
 	}
-	proposalTimeout := time.Millisecond * 100
+	proposalTimeout := time.Second
```

Comment:
Do we need this timeout in the first place? The code seems to expect that dropping must happen, so we might be ok to timeout the test if it doesn't happen.

I wonder if there is a race here which results in this proposal never being dropped (whichever the timeout is). This race could be the thing to fix then.

Reply:
> I wonder if there is a race here which results in this proposal never being dropped (whichever the timeout is). This race could be the thing to fix then.

I don't think that's the case; we see log lines indicating that the proposal was dropped when the timeout fires.

Reply:
> Do we need this timeout in the first place? The code seems to expect that dropping must happen, so we might be ok to timeout the test if it doesn't happen.

Sorry I didn't reply to this part. You're right, in that we don't strictly need the timeout. I think it just fails the test faster. I'm open to removing the thing if you prefer that approach over this one.

Reply:
Yeah, let's try removing the timeout.

The race I have in mind is:
1. `r.step = dropStep` sets the handler.
2. But before we get to `n.Propose` a couple of lines below, the `RawNode` transitions to some other state (e.g. it steps down from leadership) and resets `r.step` to a function that does not drop proposals.

I'm not sure if (2) can happen, but if it can then that would be the root cause here. We'll find this out if the tests fails again after removing the timeout.

Reply:
OTOH, if (2) was the case, then `n.Propose` would return early rather than with a timeout. It's probably just a timing issue.

Reply:
I removed the timeout. RFAL.